Makefile

```diff
@@ -153,10 +153,12 @@ test-e2e:
 	@KNUU_NAMESPACE=celestia-app E2E=true go test -mod=readonly ./test/e2e/... -timeout 30m
 .PHONY: test-e2e
 
-## test-race: Run unit tests in race mode.
+## test-race: Run tests in race mode.
 test-race:
+# TODO: Remove the -skip flag once the following tests no longer contain data races.
+# https://github.com/celestiaorg/celestia-app/issues/1369
 	@echo "--> Running tests in race mode"
-	@VERSION=$(VERSION) go test -mod=readonly -race -short ./...
+	@go test -mod=readonly ./... -v -race -skip "TestPrepareProposalConsistency|TestIntegrationTestSuite|TestQGBRPCQueries|TestSquareSizeIntegrationTest|TestStandardSDKIntegrationTestSuite|TestTxsimCommandFlags|TestTxsimCommandEnvVar|TestMintIntegrationTestSuite|TestQGBCLI|TestUpgrade"
```

Comment:
what do we think of doing something like this? 
https://github.com/celestiaorg/celestia-app/issues/1964#issuecomment-1597774436

we have a package that has an init func which modifies some global bool in a different package. then we use that bool in the tests similar to or as a replacement for t.Short()

Reply:
in this case, there would only be a single init and build flag

if we wanted to run with the race detector, then we add those build flags, which causes a single init function to change a global var

the advantage here would be that we would be able to define things in an identical way to what we do now with t.Short() in the go test itself instead of having two different apis. 

The idea of just appending something to a list feels weird, but I won't block on it

Reply:
> there would only be a single init and build flag

Wouldn't we have to add a build flag to each test file?

> in an identical way to what we do now with t.Short() in the go test itself

Will your approach lead to something like:

```diff
func TestPrepareProposalConsistency(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping TestPrepareProposalConsistency in short mode.")
	}
+	if testing.SkipRace() {
+		t.Skip("skipping TestPrepareProposalConsistency in race mode.")
+	}
```

If that's the case, it seems fine. TBH I'm not sure what you mean by single init and build flag so I'm going to merge this PR as is. Definitely open to reverting the `-skip` flag usage if there's a clearer way to do this.